src/mystery_gift.c

```diff
@@ -0,0 +1,293 @@
+#include "global.h"
+#include "mystery_gift.h"
+#include "save.h"
+#include "save_arrays.h"
+#include "constants/save_arrays.h"
+
+BOOL MysteryGiftTagIsValid(u32 tag);
+MysteryGift* SaveMysteryGift_GetByIdx(MysteryGiftSave* mg, int index);
+BOOL SaveMysteryGift_SetReceivedByIdx(MysteryGiftSave* mg, int index);
+void SaveMysteryGift_ReceivedFlagClear(MysteryGiftSave* mg, int index);
+void SaveMysteryGift_SetReceivedByCardId(MysteryGiftSave* mg, int index);
+BOOL SaveMysteryGift_TagAtIndexIsValid(const MysteryGiftSave* mg, int index);
+void SaveMysteryGift_ReceivedFlagClear(MysteryGiftSave* mg, int index);
+
+u32 Save_MysteryGift_sizeof(void) {
+    return sizeof(MysteryGiftSave);
+}
+
+void Save_MysteryGift_Init(MysteryGiftSave* mg) {
+    SaveSubstruct_UpdateCRC(SAVE_MYSTERY_GIFT);
+}
+
+BOOL MysteryGiftTagIsValid(u32 tag) {
+    return tag != MG_TAG_INVALID && tag < MG_TAG_MAX;
+}
+
+MysteryGift* SaveMysteryGift_GetByIdx(MysteryGiftSave* mg, int index) {
+    if (index >= 0 && index < NUM_SAVED_MYSTERY_GIFTS) {
+        MysteryGift* ret = &mg->gifts[index];
+        if (MysteryGiftTagIsValid(ret->tag)) {
+            return ret;
+        }
+    }
+
+    return NULL;
+}
+
+WonderCard* SaveMysteryGift_CardGetByIdx(MysteryGiftSave* mg, int index) {
+    if (index >= 0 && index < NUM_SAVED_WONDER_CARDS) {
+        WonderCard* ret = &mg->cards[index];
+        if (MysteryGiftTagIsValid(ret->tag)) {
+            return ret;
+        }
+    } else if (index == RECEIVED_WONDER_CARD_IDX) {
+        return &mg->specialWonderCard;
+    }
+
+    return NULL;
+}
+
+BOOL SaveMysteryGift_TryInsertGift(MysteryGiftSave* mg, const MysteryGift* src, int cardIdx) {
+    BOOL ret = FALSE;
+    int i;
```

Comment:
make this inline with the for loop

Reply:
applies to all instances

Reply:
This form is correct and I refuse to be told otherwise

Reply:
eh it's not super important, just being picky